tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -213,400 +352,80 @@ You can find more details about this in the Bishop book mentioned in the introdu
-
-```julia
-# Example data set - generate synthetic gene expression data
-
-# dataset available in RDatasets
-data = dataset("datasets", "iris")
-species = data[!, "Species"]
-
-# we extract the four measured quantities
-d = 4
-dat = data[!, 1:d]
-# and the number of measurements
-n = size(dat)[1];
-```
-
-First, let's look at the original data using the pPCA model.
-
-```julia
-ppca = pPCA(dat)
-
-# Here we use a different sampler, we don't always have to use NUTS:
-# Hamiltonian Monte Carlo (HMC) sampler parameters
-ϵ = 0.05
-τ = 10
-chain_ppca2 = sample(ppca, HMC(ϵ, τ), 1000)
-
-# Extract parameter estimates for plotting - mean of posterior
-w = permutedims(reshape(mean(group(chain_ppca2, :w))[:, 2], (d, d)))
-z = permutedims(reshape(mean(group(chain_ppca2, :z))[:, 2], (d, n)))'
-mu = mean(group(chain_ppca2, :m))[:, 2]
-
-X = w * z
-# X = w * z .+ mu
-
-df_rec = DataFrame(X', :auto)
-df_rec[!, :species] = species
-@vlplot(:rect, "species:o", "variable:o", color = :value)(DataFrames.stack(df_rec, 1:d))
-
-df_iris = DataFrame(z', :auto)
-rename!(df_iris, Symbol.(["z" * string(i) for i in collect(1:d)]))
-df_iris[!, :sample] = 1:n
-df_iris[!, :species] = species
-
-@vlplot(:point, x = :z1, y = :z2, color = "species:n")(df_iris)
-```
-
-We can see that the setosa species is more clearly separated from the other two species, which overlap
-considerably.
-
-We now simulate a batch effect; imagine the person taking the measurement uses two different rulers and they are slightly off.
-Again, in practice there are many different reasons for why batch effects occur and it is not always clear what is really at the basis of them,
-nor can they always be tackled via the experimental setup. So we need methods to deal with them.
-
-```julia
-## Introduce batch effect
-batch = rand(Binomial(1, 0.5), 150)
-effect = rand(Normal(2.4, 0.6), 150)
-batch_dat = dat .+ batch .* effect
-
-ppca_batch = pPCA(batch_dat)
-chain_ppcaBatch = sample(ppca_batch, HMC(ϵ, τ), 1000)
-describe(chain_ppcaBatch)[1]
-
-z = permutedims(reshape(mean(group(chain_ppcaBatch, :z))[:, 2], (d, n)))'
-df_pre = DataFrame(z', :auto)
-rename!(df_pre, Symbol.(["z" * string(i) for i in collect(1:d)]))
-df_pre[!, :sample] = 1:n
-df_pre[!, :species] = species
-df_pre[!, :batch] = batch
-
-@vlplot(:point, x = :z1, y = :z2, color = "species:n", shape = :batch)(df_pre)
-```
-
-The batch effect makes it much harder to distinguish the species. And importantly, if we are not aware of the
-batches, this might lead us to make wrong conclusions about the data.
+Instead of drawing samples of each entry in $w$ from a standard normal, this time we repeatedly draw $D$ samples from the D-dimensional MND, forming a $D \times D$ matrix $w$.
```

Comment:
```suggestion
Instead of drawing samples of each entry in $\mathbf{W}$ from a standard normal, this time we repeatedly draw $D$ samples from the D-dimensional `MvNormal`, forming a $D \times D$ matrix $\mathbf{W}$.
```